### PR TITLE
Cleanup duplicate html class

### LINKED ISSUE #28345: Incorrect CSS selector in shipment page
https://github.com/magento/magento2/blob/c9a246a100a0208ea156a3596864d638a2a0cfa3/app/code/Magento/Sales/view/frontend/templates/order/print/shipment.phtml#L64

I think this line should be like this.

`<div class="box box-order-billing-address">`

Am I wrong?

### LINKED ISSUE #29261: [Issue] Cleanup duplicate html class
This issue is automatically created based on existing pull request: magento/magento2#28639: Cleanup duplicate html class

---------

### Summary (*)

Magento 2.4-develop
Reopen pull request as request from core maintainer
This PR clean the duplicate class in template shipment

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Reopen pull request as request from core maintainer
This PR clean the duplicate class in template shipment 
### Fixed Issues (if relevant)

1. Fixes magento/magento2#28345

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#29261: Cleanup duplicate html class

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by lbajsarowicz — APPROVED** [review]
Nice catch 👍 Could you make sure that all the related selectors in functional tests are adjusted, too?

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7658](https://jira.corp.magento.com/browse/ENGCOM-7658) has been created to process this Pull Request

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Nice catch 👍 Could you make sure that all the related selectors in functional tests are adjusted, too?

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by lbajsarowicz** [comment]
@magento run Static Tests

File: app/code/Magento/Sales/view/frontend/templates/order/print/shipment.phtml [hunk 1]
```diff
@@ -3,26 +3,34 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+
+use Magento\Framework\Escaper;
```
Comment (lbajsarowicz) [inline app/code/Magento/Sales/view/frontend/templates/order/print/shipment.phtml]: We don't use imports in `phtml` files.

File: app/code/Magento/Sales/view/frontend/templates/order/print/shipment.phtml [hunk 2]
```diff
@@ -3,26 +3,34 @@
...
+
+/**
+ * @var \Magento\Sales\Block\Order\PrintOrder\Shipment $block
+ * @var Escaper $escaper
```
Comment (lbajsarowicz) [inline app/code/Magento/Sales/view/frontend/templates/order/print/shipment.phtml]: Just use FQN there

File: app/code/Magento/Sales/view/frontend/templates/order/print/shipment.phtml [hunk 3]
```diff
@@ -3,26 +3,34 @@
...
 <?php $order = $block->getOrder(); ?>
-<?php if (!$block->getObjectData($order, 'is_virtual')) : ?>
-    <?php foreach ($block->getShipmentsCollection() as $shipment) : ?>
+<?php if (!$block->getObjectData($order, 'is_virtual')): ?>
+    <?php foreach ($block->getShipmentsCollection() as $shipment): ?>
         <div class="order-details-items shipments">
             <div class="order-title">
-                <strong><?= $block->escapeHtml(__('Shipment #%1', $block->getObjectData($shipment, 'increment_id'))) ?></strong>
+                <strong><?= $escaper->escapeHtml(
+                    __('Shipment #%1', $block->getObjectData($shipment, 'increment_id'))) ?></strong>
```
Comment (lbajsarowicz) [inline app/code/Magento/Sales/view/frontend/templates/order/print/shipment.phtml]:
```suggestion
                <strong><?= $escaper->escapeHtml(
                    __('Shipment #%1', $block->getObjectData($shipment, 'increment_id')
                )) ?></strong>
```

File: app/code/Magento/Sales/view/frontend/templates/order/print/shipment.phtml [hunk 4]
```diff
@@ -3,26 +3,34 @@
...
             </div>
             <div class="table-wrapper order-items-shipment">
-                <table class="data table table-order-items shipment" id="my-shipment-table-<?= (int) $block->getObjectData($shipment, 'id') ?>">
-                    <caption class="table-caption"><?= $block->escapeHtml(__('Items Invoiced')) ?></caption>
+                <table class="data table table-order-items shipment"
+                        id="my-shipment-table-<?= (int) $block->getObjectData($shipment, 'id') ?>">
```
Comment (lbajsarowicz) [inline app/code/Magento/Sales/view/frontend/templates/order/print/shipment.phtml]:
```suggestion
                        id="my-shipment-table-<?= (int)$block->getObjectData($shipment, 'id') ?>">
```

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Please review the comments and adjust the code to comply :)
Thanks for your effort, really appreciate that you also fix the existing issues there.

**Comment by mrtuvn** [comment]
@lbajsarowicz just updated as your suggestion

**Comment by lbajsarowicz** [comment]
@magento run all tests

After tests pass - I'll approve the PR.

File: app/code/Magento/Sales/view/frontend/templates/order/print/shipment.phtml (hunk 1 above)
Comment (ihor-sviziev) [reply app/code/Magento/Sales/view/frontend/templates/order/print/shipment.phtml]: @lbajsarowicz Do we have such rule? It seems to me quite stupid. Don't you thing so?

Comment (lbajsarowicz) [reply app/code/Magento/Sales/view/frontend/templates/order/print/shipment.phtml]: I'm not sure if `stupid` is an appropriate word for what, however:
- Importing class for single annotation is - in my opinion - not necessary
- we should keep consistency.
- PHTML file is not a regular PHP file, as it's passed through the Template Engine, thus - introducing elements such as `use` might cause issues 

I'll try to find resource about this for you @ihor-sviziev

**Comment by mrtuvn** [comment]
We have all green now

**Review by lbajsarowicz — APPROVED** [review]
:heavy_check_mark:  Perfect job, thank you for introducing `$escaper` - that's a smart move!

**Comment by lbajsarowicz** [comment]
@magento run all tests

Running tests again, as Contributor squashed all the changes, so the results of previous CI were dropped.

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7658](https://jira.corp.magento.com/browse/ENGCOM-7658) has been created to process this Pull Request

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
got struggle with fix static the line 22. Very fun
attempt #1 Line indented incorrectly; expected at least 24 spaces, found 16
attempt #2 Multi-line function call not indented correctly; expected 16 spaces but found 24
i have to create new line after tag open

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed
Was checked that print shipment works as before
**Before:**
[MainOrder # 000000002.pdf](https://github.com/magento/magento2/files/4895230/MainOrder.000000002.pdf)
image

**After:**
[PROrder # 000000003.pdf](https://github.com/magento/magento2/files/4895233/PROrder.000000003.pdf)
image

**Comment by engcom-Delta** [comment]
@magento create issue

**Pull request merged**